bug(developer): lexical model compiler requires `@keymanapp/keyman-version`

### Describe the bug

Running kmlmc-16.0.94 fails with an error:

```
$ node ./node_modules/\@keymanapp/lexical-model-compiler/dist/kmlmc.js -V
node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module '@keymanapp/keyman-version'
Require stack:
- C:\Projects\keyman\lexical-models\node_modules\@keymanapp\lexical-model-compiler\dist\lexical-model-compiler\model-definitions.js
- C:\Projects\keyman\lexical-models\node_modules\@keymanapp\lexical-model-compiler\dist\lexical-model-compiler\lexical-model-compiler.js
- C:\Projects\keyman\lexical-models\node_modules\@keymanapp\lexical-model-compiler\dist\util\util.js
- C:\Projects\keyman\lexical-models\node_modules\@keymanapp\lexical-model-compiler\dist\kmlmc.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (C:\Projects\keyman\lexical-models\node_modules\@keymanapp\lexical-model-compiler\dist\lexical-model-compiler\model-definitions.js:5:24)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'C:\\Projects\\keyman\\lexical-models\\node_modules\\@keymanapp\\lexical-model-compiler\\dist\\lexical-model-compiler\\model-definitions.js',
    'C:\\Projects\\keyman\\lexical-models\\node_modules\\@keymanapp\\lexical-model-compiler\\dist\\lexical-model-compiler\\lexical-model-compiler.js',
    'C:\\Projects\\keyman\\lexical-models\\node_modules\\@keymanapp\\lexical-model-compiler\\dist\\util\\util.js',
    'C:\\Projects\\keyman\\lexical-models\\node_modules\\@keymanapp\\lexical-model-compiler\\dist\\kmlmc.js'
  ]
}
```

### Reproduce the bug

_No response_

### Expected behavior

_No response_

### Related issues

_No response_

### Keyman apps

- [ ] Keyman for Android
- [ ] Keyman for iPhone and iPad
- [ ] Keyman for Linux
- [ ] Keyman for macOS
- [ ] Keyman for Windows
- [X] Keyman Developer
- [ ] KeymanWeb
- [ ] Other - give details at bottom of form

### Keyman version

16.0.94-beta

### Operating system

_No response_

### Device

_No response_

### Target application

_No response_

### Browser

_No response_

### Keyboard name

_No response_

### Keyboard version

_No response_

### Language name

_No response_

### Additional context

_No response_

--- a/common/web/keyman-version/.npmignore
+++ b/common/web/keyman-version/.npmignore
@@ -1,0 +1,5 @@
+build/index.tsbuildinfo
+build/tsconfig.esm.tsbuildinfo
+build.sh
+index.ts
+tsconfig.json

--- a/common/web/keyman-version/build.sh
+++ b/common/web/keyman-version/build.sh
@@ -19,7 +19,12 @@ cd "$THIS_SCRIPT_PATH"
 
 ################################ Main script ################################
 
-builder_describe "Build the include script for current Keyman version" configure clean build
+builder_describe "Build the include script for current Keyman version" \
+  configure \
+  clean \
+  build \
+  publish \
+  --dry-run
 
 builder_describe_outputs \
   configure "/node_modules" \
@@ -67,4 +72,10 @@ if builder_start_action build; then
   fi
 
   builder_finish_action success build
+fi
+
+if builder_start_action publish; then
+  . "$KEYMAN_ROOT/resources/build/build-utils-ci.inc.sh"
+  builder_publish_to_npm
+  builder_finish_action success publish
 fi

--- a/developer/src/kmlmc/.npmignore
+++ b/developer/src/kmlmc/.npmignore
@@ -1,6 +1,9 @@
 # Ignore files required only in development.
-build.sh
-bundle.sh
+dist-tests/*
 source/*
 tests/*
+build.sh
+bundle.sh
+Makefile
 tsconfig.json
+tsconfig.tsbuildinfo

--- a/developer/src/kmlmc/build.sh
+++ b/developer/src/kmlmc/build.sh
@@ -120,4 +120,9 @@ if (( should_publish )); then
   # See `npm help publish` for more details.
   echo "Publishing $DRY_RUN npm package with tag $npm_dist_tag"
   npm publish $DRY_RUN --access public --tag $npm_dist_tag || fail "Could not publish $npm_dist_tag release."
+
+  # For now, kmlmc will have responsibility for publishing keyman-version. In
+  # the future, we should probably have a top-level npm publish script that
+  # publishes all modules for a given release version
+  "$KEYMAN_ROOT/common/web/keyman-version/build.sh" publish $DRY_RUN
 fi

--- a/resources/build/build-utils-ci.inc.sh
+++ b/resources/build/build-utils-ci.inc.sh
@@ -60,3 +60,44 @@ function builder_pull_has_label() {
   fi
   return 1
 }
+
+#
+# Publishes the package in `cwd` to npm
+#
+# If the `--dry-run` option is available and specified as a command-line
+# parameter, will do a dry run
+#
+# Note that `package.json` will be dirty after this command, as the `version`
+# field will be added to it. This change should not be committed to the
+# repository.
+#
+# Usage:
+# ```bash
+#   builder_publish_to_npm
+# ```
+#
+function builder_publish_to_npm() {
+  local dist_tag=$TIER dry_run
+
+  if [[ $TIER == stable ]]; then
+    dist_tag=latest
+  fi
+
+  if builder_has_option --dry-run; then
+    dry_run=--dry-run
+  fi
+
+  # We use --no-git-tag-version because our CI system controls version numbering and
+  # already tags releases. We also want to have the version of this match the
+  # release of Keyman Developer -- these two versions should be in sync. Because this
+  # is a large repo with multiple projects and build systems, it's better for us that
+  # individual build systems don't take too much ownership of git tagging. :)
+  npm version --allow-same-version --no-git-tag-version --no-commit-hooks "$VERSION_WITH_TAG"
+
+  # Note: In either case, npm publish MUST be given --access public to publish
+  # a package in the @keymanapp scope on the public npm package index.
+  #
+  # See `npm help publish` for more details.
+  echo "Publishing $dry_run npm package $THIS_SCRIPT_IDENTIFIER with tag $dist_tag"
+  npm publish $dry_run --access public --tag $dist_tag
+}

